Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -2349,8 +2349,17 @@ void RenderBlock::computeBlockPreferredLogicalWidths(LayoutUnit& minLogicalWidth
         // A margin basically has three types: fixed, percentage, and auto (variable).
         // Auto and percentage margins simply become 0 when computing min/max width.
         // Fixed margins can be added in as is.
-        Length startMarginLength = childStyle.marginStartUsing(&styleToUse);
-        Length endMarginLength = childStyle.marginEndUsing(&styleToUse);
+        Length startMarginLength;
+        Length endMarginLength;
+        if (auto blockBox = dynamicDowncast<RenderBlockFlow>(this)) {
```

Comment:
why is this limited to RenderBlockFlow -any particular reason why shouldChildInlineMarginContributeToContainerIntrinsicSize is on RenderBlockFlow?

Reply:
I added it to that because of this part of the spec: 

`It also affects floats for which the block container is a containing block by: Zeroing the inline-axis margins of a float for the purpose of calculating its intrinsic size contributions or its size in the container’s inline axis.`

But maybe the interface here could be a little better. One thing we could do is maybe make this a virtual method inside `RenderBox` and have the `RenderBlockFlow` version override it?

Reply:
It's fine, it's just odd that you call a base class function and immediately downcast to the derived class you called from (and yes, it should be a virtual function on the RenderBlock and RenderBlockFlow would override it).